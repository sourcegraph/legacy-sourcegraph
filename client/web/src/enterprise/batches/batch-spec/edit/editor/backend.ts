import { gql } from '@sourcegraph/http-client'

export const BATCH_SPEC_EXECUTION_AVAILABLE_SECRET_KEYS = gql`
    query UserBatchSpecExecutionAvailableSecretKeys($user: ID!) {
        node(id: $user) {
            __typename
            ... on User {
                executorSecrets(scope: BATCHES, first: 1000) {
                    nodes {
                        key
                    }
                }
            }
        }
    }
`
