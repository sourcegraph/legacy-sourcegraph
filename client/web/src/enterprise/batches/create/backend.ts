import { gql } from '@sourcegraph/shared/src/graphql/graphql'

export const EXECUTE_BATCH_SPEC = gql`
    mutation ExecuteBatchSpec($batchSpec: ID!) {
        executeBatchSpec(batchSpec: $batchSpec) {
            id
            namespace {
                url
            }
        }
    }
`

export const CREATE_BATCH_SPEC_FROM_RAW = gql`
    mutation CreateBatchSpecFromRaw($spec: String!, $namespace: ID!) {
        createBatchSpecFromRaw(batchSpec: $spec, namespace: $namespace) {
            id
        }
    }
`

export const REPLACE_BATCH_SPEC_INPUT = gql`
    mutation ReplaceBatchSpecInput($previousSpec: ID!, $spec: String!) {
        replaceBatchSpecInput(previousSpec: $previousSpec, batchSpec: $spec) {
            id
        }
    }
`

export const WORKSPACE_RESOLUTION_STATUS = gql`
    query WorkspaceResolutionStatus($batchSpec: ID!) {
        node(id: $batchSpec) {
            __typename
            ... on BatchSpec {
                workspaceResolution {
                    state
                    failureMessage
                }
            }
        }
    }
`

export const WORKSPACES_AND_IMPORTING_CHANGESETS = gql`
    query WorkspacesAndImportingChangesets($batchSpec: ID!) {
        node(id: $batchSpec) {
            __typename
            ... on BatchSpec {
                workspaceResolution {
                    workspaces(first: 10000) {
                        nodes {
                            repository {
                                id
                                name
                                url
                            }
                            ignored
                            unsupported
                            branch {
                                id
                                abbrevName
                                displayName
                                target {
                                    oid
                                }
                            }
                            path
                            onlyFetchWorkspace
                            steps {
                                run
                                container
                            }
                            searchResultPaths
                            cachedResultFound
                        }
                    }
                }
                importingChangesets(first: 10000) {
                    totalCount
                    nodes {
                        __typename
                        id
                        ... on VisibleChangesetSpec {
                            description {
                                __typename
                                ... on ExistingChangesetReference {
                                    baseRepository {
                                        name
                                        url
                                    }
                                    externalID
                                }
                            }
                        }
                    }
                }
            }
        }
    }
`
