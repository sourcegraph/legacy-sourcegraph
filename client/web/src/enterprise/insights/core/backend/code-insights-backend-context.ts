import React from 'react'
import { throwError } from 'rxjs'

import { CodeInsightsBackend } from './code-insights-backend'

const errorMockMethod = (methodName: string) => () => throwError(new Error(`Implement ${methodName} method first`))

export class FakeDefaultCodeInsightsBackend implements CodeInsightsBackend {
    // Insights
    public getInsights = errorMockMethod('getInsights')
    public getInsightById = errorMockMethod('getInsightById')
    public findInsightByName = errorMockMethod('findInsightByName')
    public getReachableInsights = errorMockMethod('getReachableInsights')
    public getBackendInsightData = errorMockMethod('getBackendInsightData')
    public getBuiltInInsightData = errorMockMethod('getBuiltInInsightData')
    public getInsightSubjects = errorMockMethod('getInsightSubjects')
    public getSubjectSettingsById = errorMockMethod('getSubjectSettingsById')
    public createInsight = errorMockMethod('createInsight')
    public createInsightWithNewFilters = errorMockMethod('createInsightWithNewFilters')
    public updateInsight = errorMockMethod('updateInsight')
    public deleteInsight = errorMockMethod('deleteInsight')

    // Dashboards
    public getDashboards = errorMockMethod('getDashboards')
    public getDashboardById = errorMockMethod('getDashboardById')
    public findDashboardByName = errorMockMethod('findDashboardByName')
    public createDashboard = errorMockMethod('createDashboard')
    public deleteDashboard = errorMockMethod('deleteDashboard')
    public updateDashboard = errorMockMethod('updateDashboard')

    // Live preview fetchers
    public getSearchInsightContent = () => errorMockMethod('getSearchInsightContent')().toPromise()
    public getLangStatsInsightContent = () => errorMockMethod('getLangStatsInsightContent')().toPromise()

    // Repositories API
    public getRepositorySuggestions = () => errorMockMethod('getRepositorySuggestions')().toPromise()
    public getResolvedSearchRepositories = () => errorMockMethod('getResolvedSearchRepositories')().toPromise()
}

export const CodeInsightsBackendContext = React.createContext<CodeInsightsBackend>(new FakeDefaultCodeInsightsBackend())
