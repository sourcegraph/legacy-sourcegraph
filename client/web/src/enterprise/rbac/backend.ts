import { dataOrThrowErrors } from '@sourcegraph/http-client'
import { useShowMorePagination } from '../../components/FilteredConnection/hooks/useShowMorePagination'

const PERMISSIONS_FRAGMENT = gql`
    fragment Permission on Permission {
        id
        namespace
        action
    }
`

const ROLES_WITH_PERMISSIONS_FRAGMENT = gql`
    fragment RolesWithPermissions on Role {
        id
        name
        system
    }
`

export const ROLES_QUERY = gql`
    query RolsList($first: Int, $after: String) {
        roles(first: $first, after: $after) {
            totalCount
            pageInfo {
                hasNextPage
                endCursor
            }
            nodes {
                ...RolesWithPermissions
            }
        }
    }

    ${ROLES_WITH_PERMISSIONS_FRAGMENT}
`

export const useRolesConnection = () => useShowMorePagination({
    query: ROLES_QUERY,
    variables: {
        first: 20,
        after: null,
    },
    getConnection: result => {
        const { roles } = dataOrThrowErrors(result)
        return roles
    }
})
