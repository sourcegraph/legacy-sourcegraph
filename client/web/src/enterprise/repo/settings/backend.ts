import { gql, dataOrThrowErrors } from '@sourcegraph/http-client'

export const RepoPermissionsInfoQuery = gql`
    query RepoPermissionsInfo($repoID: ID!, $first: Int, $last: Int, $after: String, $before: String, $query: String) {
        node(id: $repoID) {
            __typename
            ... on Repository {
                ...RepoPermissionsInfoRepoNode
            }
        }
    }

    fragment RepoPermissionsInfoRepoNode on Repository {
        permissionsInfo {
            syncedAt
            updatedAt
            unrestricted
            users(first: $first, last: $last, after: $after, before: $before, query: $query) {
                nodes {
                    ...PermissionsInfoUserFields
                }
                totalCount
                pageInfo {
                    hasNextPage
                    hasPreviousPage
                    startCursor
                    endCursor
                }
            }
        }
    }

    fragment PermissionsInfoUserFields on PermissionsInfoUserNode {
        id
        reason
        updatedAt
        user {
            id
            username
            displayName
            email
            avatarURL
        }
    }
`

export const useExternalServicesConnection = (
    vars: ExternalServicesVariables
): UseShowMorePaginationResult<ExternalServicesResult, ListExternalServiceFields> =>
    useShowMorePagination<ExternalServicesResult, ExternalServicesVariables, ListExternalServiceFields>({
        query: EXTERNAL_SERVICES,
        variables: { after: vars.after, first: vars.first ?? 10 },
        getConnection: result => {
            const { externalServices } = dataOrThrowErrors(result)
            return externalServices
        },
        options: {
            fetchPolicy: 'cache-and-network',
            pollInterval: 15000,
        },
        getConnection: result => {
            const { node } = dataOrThrowErrors(result)

            if (!node) {
                throw new Error('Repository not found')
            }
        },
    })
