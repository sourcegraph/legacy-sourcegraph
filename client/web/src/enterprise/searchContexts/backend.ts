import { ApolloQueryResult, gql, ApolloClient } from '@apollo/client'

import type { InputMaybe, RepositoriesByNamesResult, RepositoriesByNamesVariables } from '../../graphql-operations'

const query = gql`
    query RepositoriesByNames($names: [String!]!, $first: Int!, $after: String) {
        repositories(names: $names, first: $first, after: $after) {
            pageInfo {
                hasNextPage
                endCursor
            }
            nodes {
                id
                name
            }
        }
    }
`
export async function fetchRepositoriesByNames(
    names: string[],
    apolloClient: ApolloClient<object>
): Promise<RepositoriesByNamesResult['repositories']['nodes']> {
    let repos: RepositoriesByNamesResult['repositories']['nodes'] = []
    const first = names.length
    let after: InputMaybe<string> = null

    while (true) {
        const result: ApolloQueryResult<RepositoriesByNamesResult> = await apolloClient.query<
            RepositoriesByNamesResult,
            RepositoriesByNamesVariables
        >({
            query,
            variables: {
                names,
                first,
                after,
            },
        })

        if (result.error) {
            throw new Error(result.error.message)
        }

        repos = repos.concat(result.data.repositories.nodes)
        if (!result.data.repositories.pageInfo.hasNextPage) {
            break
        }
        after = result.data.repositories.pageInfo.endCursor
    }
    return repos
}
