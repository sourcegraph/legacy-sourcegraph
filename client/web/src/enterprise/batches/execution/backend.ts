import { Observable } from 'rxjs'
import { map } from 'rxjs/operators'

import { dataOrThrowErrors, gql } from '@sourcegraph/shared/src/graphql/graphql'

import { requestGraphQL } from '../../../backend/graphql'
import {
    BatchSpecExecutionByIDResult,
    BatchSpecExecutionByIDVariables,
    BatchSpecExecutionFields,
    CancelBatchSpecExecutionResult,
    CancelBatchSpecExecutionVariables,
    Scalars,
} from '../../../graphql-operations'

const batchSpecExecutionFieldsFragment = gql`
    fragment BatchSpecExecutionFields on BatchSpec {
        id
        originalInput
        state
        createdAt
        startedAt
        finishedAt
        failureMessage
        applyURL
        creator {
            id
            url
            displayName
        }
        namespace {
            id
            url
            namespaceName
        }
        workspaceResolution {
            workspaces {
                totalCount
                pageInfo {
                    endCursor
                    hasNextPage
                }
                nodes {
                    id
                    steps {
                        run
                        diffStat {
                            added
                            changed
                            deleted
                        }
                        diff {
                            baseRepository {
                                name
                            }
                        }
                        container
                        cachedResultFound
                        skipped
                        outputLines
                        startedAt
                        finishedAt
                        exitCode
                        environment {
                            name
                            value
                        }
                        outputVariables {
                            name
                            value
                        }
                    }
                    searchResultPaths
                    startedAt
                    finishedAt
                    failureMessage
                    state
                    changesetSpecs {
                        id
                        type
                        __typename
                        ... on VisibleChangesetSpec {
                            description {
                                ... on GitBranchChangesetDescription {
                                    title
                                }
                                ... on ExistingChangesetReference {
                                    externalID
                                }
                            }
                        }
                    }
                    placeInQueue
                    repository {
                        name
                        url
                    }
                    branch {
                        name
                    }
                    path
                    onlyFetchWorkspace
                    ignored
                    unsupported
                    cachedResultFound
                    stages {
                        setup {
                            key
                            command
                            startTime
                            exitCode
                            out
                            durationMilliseconds
                        }
                        srcExec {
                            key
                            command
                            startTime
                            exitCode
                            out
                            durationMilliseconds
                        }
                        teardown {
                            key
                            command
                            startTime
                            exitCode
                            out
                            durationMilliseconds
                        }
                    }
                }
            }
        }
    }
`

export const fetchBatchSpecExecution = (id: Scalars['ID']): Observable<BatchSpecExecutionFields | null> =>
    requestGraphQL<BatchSpecExecutionByIDResult, BatchSpecExecutionByIDVariables>(
        gql`
            query BatchSpecExecutionByID($id: ID!) {
                node(id: $id) {
                    __typename
                    ...BatchSpecExecutionFields
                }
            }
            ${batchSpecExecutionFieldsFragment}
        `,
        { id }
    ).pipe(
        map(dataOrThrowErrors),
        map(({ node }) => {
            if (!node) {
                return null
            }
            if (node.__typename !== 'BatchSpec') {
                throw new Error(`Node is a ${node.__typename}, not a BatchSpec`)
            }
            return node
        })
    )

export async function cancelBatchSpecExecution(id: Scalars['ID']): Promise<BatchSpecExecutionFields> {
    const result = await requestGraphQL<CancelBatchSpecExecutionResult, CancelBatchSpecExecutionVariables>(
        gql`
            mutation CancelBatchSpecExecution($id: ID!) {
                cancelBatchSpecExecution(batchSpec: $id) {
                    ...BatchSpecExecutionFields
                }
            }

            ${batchSpecExecutionFieldsFragment}
        `,
        { id }
    ).toPromise()
    return dataOrThrowErrors(result).cancelBatchSpecExecution
}
