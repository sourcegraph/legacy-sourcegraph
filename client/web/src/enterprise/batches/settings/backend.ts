import { Observable } from 'rxjs'
import { map, mapTo } from 'rxjs/operators'

import { dataOrThrowErrors, gql } from '@sourcegraph/http-client'

import { requestGraphQL } from '../../../backend/graphql'
import {
    BatchChangesCodeHostFields,
    BatchChangesCodeHostsFields,
    BatchChangesCredentialFields,
    CreateBatchChangesCredentialResult,
    CreateBatchChangesCredentialVariables,
    DeleteBatchChangesCredentialResult,
    DeleteBatchChangesCredentialVariables,
    GlobalBatchChangesCodeHostsResult,
    GlobalBatchChangesCodeHostsVariables,
    Scalars,
    UserBatchChangesCodeHostsResult,
    UserBatchChangesCodeHostsVariables,
} from '../../../graphql-operations'
import { useConnection, UseConnectionResult } from '../../../components/FilteredConnection/hooks/useConnection'

export const batchChangesCredentialFieldsFragment = gql`
    fragment BatchChangesCredentialFields on BatchChangesCredential {
        id
        sshPublicKey
        isSiteCredential
    }
`

export function createBatchChangesCredential(
    args: CreateBatchChangesCredentialVariables
): Promise<BatchChangesCredentialFields> {
    return requestGraphQL<CreateBatchChangesCredentialResult, CreateBatchChangesCredentialVariables>(
        gql`
            mutation CreateBatchChangesCredential(
                $user: ID
                $credential: String!
                $externalServiceKind: ExternalServiceKind!
                $externalServiceURL: String!
            ) {
                createBatchChangesCredential(
                    user: $user
                    credential: $credential
                    externalServiceKind: $externalServiceKind
                    externalServiceURL: $externalServiceURL
                ) {
                    ...BatchChangesCredentialFields
                }
            }

            ${batchChangesCredentialFieldsFragment}
        `,
        args
    )
        .pipe(
            map(dataOrThrowErrors),
            map(data => data.createBatchChangesCredential)
        )
        .toPromise()
}

export function deleteBatchChangesCredential(id: Scalars['ID']): Promise<void> {
    return requestGraphQL<DeleteBatchChangesCredentialResult, DeleteBatchChangesCredentialVariables>(
        gql`
            mutation DeleteBatchChangesCredential($id: ID!) {
                deleteBatchChangesCredential(batchChangesCredential: $id) {
                    alwaysNil
                }
            }
        `,
        { id }
    )
        .pipe(map(dataOrThrowErrors), mapTo(undefined))
        .toPromise()
}

const CODE_HOST_FIELDS_FRAGMENT = gql`
    fragment BatchChangesCodeHostsFields on BatchChangesCodeHostConnection {
        totalCount
        pageInfo {
            hasNextPage
            endCursor
        }
        nodes {
            ...BatchChangesCodeHostFields
        }
    }

    fragment BatchChangesCodeHostFields on BatchChangesCodeHost {
        externalServiceKind
        externalServiceURL
        requiresSSH
        credential {
            ...BatchChangesCredentialFields
        }
    }

    ${batchChangesCredentialFieldsFragment}
`

const USER_CODE_HOSTS_QUERY = gql`
    query UserBatchChangesCodeHosts($user: ID!, $first: Int, $after: String) {
        node(id: $user) {
            __typename
            ... on User {
                batchChangesCodeHosts(first: $first, after: $after) {
                    ...BatchChangesCodeHostsFields
                }
            }
        }
    }

    ${CODE_HOST_FIELDS_FRAGMENT}
`

export const useUserBatchChangesCodeHostConnection = (
    user: Scalars['ID']
): UseConnectionResult<BatchChangesCodeHostFields> =>
    useConnection<UserBatchChangesCodeHostsResult, UserBatchChangesCodeHostsVariables, BatchChangesCodeHostFields>({
        query: USER_CODE_HOSTS_QUERY,
        variables: {
            user,
            after: null,
            first: 15,
        },
        options: {
            fetchPolicy: 'no-cache',
        },
        getConnection: result => {
            const { node } = dataOrThrowErrors(result)

            if (!node) {
                throw new Error('User not found')
            }
            if (node.__typename !== 'User') {
                throw new Error(`Node is a ${node.__typename}, not a User`)
            }
            if (!node.batchChangesCodeHosts) {
                throw new Error('No code hosts found')
            }

            return node.batchChangesCodeHosts
        },
    })

export const queryUserBatchChangesCodeHosts = ({
    user,
    first,
    after,
}: UserBatchChangesCodeHostsVariables): Observable<BatchChangesCodeHostsFields> =>
    requestGraphQL<UserBatchChangesCodeHostsResult, UserBatchChangesCodeHostsVariables>(gql``, {
        user,
        first,
        after,
    }).pipe(
        map(dataOrThrowErrors),
        map(data => {
            if (data.node === null) {
                throw new Error('User not found')
            }
            if (data.node.__typename !== 'User') {
                throw new Error(`Node is a ${data.node.__typename}, not a User`)
            }
            return data.node.batchChangesCodeHosts
        })
    )

const GLOBAL_CODE_HOSTS_QUERY = gql`
    query GlobalBatchChangesCodeHosts($first: Int, $after: String) {
        batchChangesCodeHosts(first: $first, after: $after) {
            ...BatchChangesCodeHostsFields
        }
    }

    ${CODE_HOST_FIELDS_FRAGMENT}
`

export const useGlobalBatchChangesCodeHostConnection = (): UseConnectionResult<BatchChangesCodeHostFields> =>
    useConnection<GlobalBatchChangesCodeHostsResult, GlobalBatchChangesCodeHostsVariables, BatchChangesCodeHostFields>({
        query: GLOBAL_CODE_HOSTS_QUERY,
        variables: {
            after: null,
            first: 20,
        },
        options: {
            useURL: true,
            fetchPolicy: 'no-cache',
        },
        getConnection: result => {
            const data = dataOrThrowErrors(result)

            if (!data.batchChangesCodeHosts) {
                throw new Error('No code hosts found')
            }

            return data.batchChangesCodeHosts
        },
    })

export const queryGlobalBatchChangesCodeHosts = ({
    first,
    after,
}: GlobalBatchChangesCodeHostsVariables): Observable<BatchChangesCodeHostsFields> =>
    requestGraphQL<GlobalBatchChangesCodeHostsResult, GlobalBatchChangesCodeHostsVariables>(gql``, {
        first,
        after,
    }).pipe(
        map(dataOrThrowErrors),
        map(data => data.batchChangesCodeHosts)
    )
