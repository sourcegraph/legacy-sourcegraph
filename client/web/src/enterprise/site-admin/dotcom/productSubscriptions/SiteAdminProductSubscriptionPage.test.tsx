import { MockedResponse } from '@apollo/client/testing'

import { getDocumentNode } from '@sourcegraph/http-client'
import { MockedTestProvider, waitForNextApolloResponse } from '@sourcegraph/shared/src/testing/apollo'
import { renderWithBrandedContext } from '@sourcegraph/wildcard/src/testing'

import {
    DotComProductSubscriptionResult,
    DotComProductSubscriptionVariables,
    ProductLicensesResult,
    ProductLicensesVariables,
} from '../../../../graphql-operations'

import { PRODUCT_LICENSES, DOTCOM_PRODUCT_SUBSCRIPTION } from './backend'
import { SiteAdminProductSubscriptionPage } from './SiteAdminProductSubscriptionPage'
import { mockLicenseContext } from './testUtils'

jest.mock('mdi-react/ArrowLeftIcon', () => 'ArrowLeftIcon')

jest.mock('mdi-react/AddIcon', () => 'AddIcon')

const subscriptionMock: MockedResponse<DotComProductSubscriptionResult, DotComProductSubscriptionVariables> = {
    request: {
        query: getDocumentNode(DOTCOM_PRODUCT_SUBSCRIPTION),
        variables: { uuid: '' },
    },
    result: {
        data: {
            __typename: 'Query',
            dotcom: {
                __typename: 'DotcomQuery',
                productSubscription: {
                    __typename: 'ProductSubscription',
                    createdAt: '2023-05-05T13:10:30.080Z',
                    url: '/s',
                    account: null,
                    id: 'l1',
                    isArchived: false,
                    name: 'sn1',
                    productLicenses: {
                        __typename: 'ProductLicenseConnection',
                        nodes: [
                            {
                                __typename: 'ProductLicense',
                                createdAt: '2023-05-05T13:10:30.080Z',
                                id: 'l1',
                                licenseKey: 'lk1',
                                info: {
                                    __typename: 'ProductLicenseInfo',
                                    expiresAt: '2024-05-05T13:10:30.080Z',
                                    tags: ['a'],
                                    userCount: 123,
                                },
                            },
                        ],
                        totalCount: 1,
                        pageInfo: { __typename: 'PageInfo', hasNextPage: false },
                    },
                    activeLicense: null,
                    sourcegraphAccessToken: '123',
                    llmProxyAccess: {
                        __typename: 'LLMProxyAccess',
                        enabled: false,
                        rateLimit: null,
                    },
                },
            },
        },
    },
}

const licensesMock: MockedResponse<ProductLicensesResult> = {
    request: {
        query: getDocumentNode(PRODUCT_LICENSES),
        variables: {
            first: 20,
            subscriptionUUID: '',
            after: null,
        },
    },
    result: {
        data: {
            __typename: 'Query',
            dotcom: {
                __typename: 'DotcomQuery',
                productSubscription: {
                    __typename: 'ProductSubscription',
                    productLicenses: {
                        __typename: 'ProductLicenseConnection',
                        nodes: [
                            {
                                __typename: 'ProductLicense',
                                createdAt: '2023-05-05T13:10:30.080Z',
                                id: 'l1',
                                licenseKey: 'lk1',
                                info: {
                                    __typename: 'ProductLicenseInfo',
                                    expiresAt: '2024-05-05T13:10:30.080Z',
                                    productNameWithBrand: 'NB',
                                    tags: ['a'],
                                    userCount: 123,
                                },
                                subscription: {
                                    __typename: 'ProductSubscription',
                                    id: 'l1',
                                    name: 'sn1',
                                    urlForSiteAdmin: null,
                                    account: null,
                                    activeLicense: { __typename: 'ProductLicense', id: 'l1' },
                                },
                            },
                        ],
                        totalCount: 1,
                        pageInfo: { __typename: 'PageInfo', hasNextPage: false },
                    },
                },
            },
        },
    },
}

describe('SiteAdminProductSubscriptionPage', () => {
    const origContext = window.context
    beforeEach(() => {
        window.context = mockLicenseContext
    })
    afterEach(() => {
        window.context = origContext
    })
    test('renders', async () => {
        const component = renderWithBrandedContext(
            <MockedTestProvider mocks={[subscriptionMock, licensesMock, subscriptionMock, licensesMock]}>
                <SiteAdminProductSubscriptionPage />
            </MockedTestProvider>,
            { route: '/p' }
        )
        await waitForNextApolloResponse()
        await waitForNextApolloResponse()
        expect(component.asFragment()).toMatchSnapshot()
    })
})
