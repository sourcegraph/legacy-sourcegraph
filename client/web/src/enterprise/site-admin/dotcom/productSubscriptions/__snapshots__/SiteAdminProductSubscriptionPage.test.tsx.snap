// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteAdminProductSubscriptionPage renders 1`] = `
<DocumentFragment>
  <div
    aria-live="assertive"
    class="alert alertDanger my-2"
    role="alert"
  >
     
    <span
      class="markdown"
    >
      No more mocked responses for the query: query DotComProductSubscription($uuid: String!) {
      <br />
        dotcom {
      <br />
          productSubscription(uuid: $uuid) {
      <br />
            id
      <br />
            name
      <br />
            account {
      <br />
              ...DotComProductSubscriptionEmailFields
      <br />
              __typename
      <br />
            }
      <br />
            productLicenses {
      <br />
              nodes {
      <br />
                id
      <br />
                info {
      <br />
                  tags
      <br />
                  userCount
      <br />
                  expiresAt
      <br />
                  __typename
      <br />
                }
      <br />
                licenseKey
      <br />
                createdAt
      <br />
                __typename
      <br />
              }
      <br />
              totalCount
      <br />
              pageInfo {
      <br />
                hasNextPage
      <br />
                __typename
      <br />
              }
      <br />
              __typename
      <br />
            }
      <br />
            activeLicense {
      <br />
              id
      <br />
              info {
      <br />
                productNameWithBrand
      <br />
                tags
      <br />
                userCount
      <br />
                expiresAt
      <br />
                __typename
      <br />
              }
      <br />
              licenseKey
      <br />
              createdAt
      <br />
              __typename
      <br />
            }
      <br />
            sourcegraphAccessToken
      <br />
            llmProxyAccess {
      <br />
              ...LLMProxyAccessFields
      <br />
              __typename
      <br />
            }
      <br />
            createdAt
      <br />
            isArchived
      <br />
            url
      <br />
            __typename
      <br />
          }
      <br />
          __typename
      <br />
        }
      <br />
      }
      <p />
      

      <p>
        fragment LLMProxyAccessFields on LLMProxyAccess {
        <br />
          enabled
        <br />
          rateLimit {
        <br />
            ...LLMProxyRateLimitFields
        <br />
            __typename
        <br />
          }
        <br />
          __typename
        <br />
        }
      </p>
      

      <p>
        fragment LLMProxyRateLimitFields on LLMProxyRateLimit {
        <br />
          source
        <br />
          limit
        <br />
          intervalSeconds
        <br />
          __typename
        <br />
        }
      </p>
      

      <p>
        fragment DotComProductSubscriptionEmailFields on User {
        <br />
          id
        <br />
          username
        <br />
          displayName
        <br />
          emails {
        <br />
            email
        <br />
            verified
        <br />
            __typename
        <br />
          }
        <br />
          __typename
        <br />
        }
      </p>
      

      <p>
        Expected variables: {"uuid":""}
      </p>
    </span>
  </div>
</DocumentFragment>
`;
